### PR TITLE
Add unit test for `createModeledMethodKey`

### PR DESCRIPTION
**Pull request by shati-patel**
Follow up to https://github.com/github/vscode-codeql/pull/3518 to add some tests! 🧪 

I'm not really sure what else to test 🤔 Suggestions welcome ❤️ 

## Checklist

N/A, no user impact

- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

File: extensions/ql-vscode/test/unit-tests/model-editor/modeled-method.test.ts [hunk 1]
```diff
@@ -50,3 +54,27 @@ describe("modeledMethodSupportsProvenance", () => {
...
+  it("should check that two modeled methods are the same", () => {
+    const modeledMethod = createSummaryModeledMethod();
+    const key = createModeledMethodKey(modeledMethod);
+
+    const modeledMethod2 = createSummaryModeledMethod();
+    const key2 = createModeledMethodKey(modeledMethod2);
+
+    // Object references are different, but the keys are the same.
+    expect(modeledMethod === modeledMethod2).toBe(false);
+    expect(key === key2).toBe(true);
```
Comment (koesie10) [inline extensions/ql-vscode/test/unit-tests/model-editor/modeled-method.test.ts:78]: This should result in better error messages:

```suggestion
    expect(modeledMethod).not.toBe(modeledMethod2);
    expect(key).toEqual(key2);
```

**Review by koesie10 — COMMENTED** [review]
Thanks! 

For additional tests, I think you might be able to get some inspiration from the [`validation.test.ts` unit tests](https://github.com/github/vscode-codeql/blob/98b0850f684dcf8f98682bf6ec02135597f5de13/extensions/ql-vscode/test/unit-tests/model-editor/shared/validation.test.ts):
- [A test that checks that the `provenance` field is always set to `manual`, regardless of the actual provenance](https://github.com/github/vscode-codeql/blob/98b0850f684dcf8f98682bf6ec02135597f5de13/extensions/ql-vscode/test/unit-tests/model-editor/shared/validation.test.ts#L97-L104)
- [A test that checks that "unused" fields (e.g. an `input` field on a `source` model) are ignored](https://github.com/github/vscode-codeql/blob/98b0850f684dcf8f98682bf6ec02135597f5de13/extensions/ql-vscode/test/unit-tests/model-editor/shared/validation.test.ts#L118-L131). It might be useful to repeat this for `source`, `sink`, `summary` and `neutral`, like in `validation.test.ts`.

**Comment by shati-patel** [comment]
Thank you for the pointers, @koesie10! I've added some more tests 🧪

**Review by koesie10 — APPROVED** [review]
Thanks!

**Pull request merged**